editoast/src/views/infra/mod.rs

```diff
@@ -561,6 +565,79 @@ pub mod tests {
         assert!(Infra::delete(db_pool, cloned_infra_id).await.unwrap());
     }
 
+    #[derive(QueryableByName)]
+    struct Count {
+        #[diesel(sql_type = BigInt)]
+        nb: i64,
+    }
+
+    #[rstest]
+    async fn infra_clone(db_pool: Data<DbPool>) {
+        let app = create_test_service().await;
+        let small_infra = &small_infra(db_pool.clone()).await.model;
+        let small_infra_id = small_infra.id.unwrap();
+
+        let pg_config = PostgresConfig::default();
+        let pg_config_url = pg_config.url().expect("cannot get postgres config url");
+        let mut conn = PgConnection::establish(pg_config_url.as_str())
+            .await
+            .expect("Error while connecting DB");
+
+        let infra_cache = InfraCache::load(&mut conn, small_infra).await.unwrap();
+
+        generated_data::refresh_all(db_pool.clone(), small_infra_id, &infra_cache)
+            .await
+            .unwrap();
+
+        let errors_count =
+            sql_query("SELECT COUNT (*) as nb from infra_layer_error where infra_id = $1")
+                .bind::<BigInt, _>(small_infra_id)
+                .get_result::<Count>(&mut conn)
+                .await
+                .unwrap();
+        assert!(errors_count.nb > 1);
+
+        let req_clone = TestRequest::post()
+            .uri(format!("/infra/{}/clone/?name=cloned_infra", small_infra_id).as_str())
+            .to_request();
+        let response = call_service(&app, req_clone).await;
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let cloned_infra_id: i64 = read_body_json(response).await;
+
+        for object in ObjectType::iter() {
+            let model_table = object.get_table();
+
+            let mut vec_table = vec![model_table];
+
+            if let Some(layer_table) = object.get_geometry_layer_table() {
+                vec_table.push(layer_table);
+            }
```

Comment:
I would split the table list construction and the check of duplication.

```suggestion
       let mut tables = vec!["infra_layer_error"];
        for object in ObjectType::iter() {
            tables.push(object.get_table());
            if let Some(layer_table) = object.get_geometry_layer_table() {
                tables.push(layer_table);
            }
        }
```